File: Data Structures & Algorithms/is-palindrome/submission-1.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        StringBuilder sb = new StringBuilder();
        String strs[] = s.split(" ");
        for(String str : strs){
            for(char ch : str.toCharArray()){
                if((ch>='a' && ch<='z')||(ch>='0' && ch<='9')){
                    sb.append(ch);
                }
            }
        }
        return helper(sb.toString());
    }
    public boolean helper(String str){
        int i = 0;
        int j = str.length()-1;
        while(i<j){
            if(str.charAt(i)!=str.charAt(j)) return false;
            i++;
            j--;
        }
        return true;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-1.py
class Solution:
    def helper(self,s,strs):
        for i in range(len(s)):
            if s[i] != strs[i]:
                return s[:i]
        return s
    def longestCommonPrefix(self, strs: List[str]) -> str:
        m = float('inf')
        s = ''
        for sk in strs:
            if len(sk)<m:
                s = sk
                m = len(s)
        for sk in strs:
            s = self.helper(s,sk)
        return s